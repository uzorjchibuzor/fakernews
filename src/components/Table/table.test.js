import React from 'react';
import ReactDOM from 'react-dom'
import Table from './table';
import renderer from 'react-test-renderer'

describe('Table', () => {
  const props = {
    list: [
    { title: '1', author: '1', num_comments: 1, points: 2, objectID: 'y' },
    { title: '2', author: '2', num_comments: 1, points: 2, objectID: 'z' }],
  }

    const dummyFunc = () => 1

  it('renders', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Table  list={props.list} onDismiss={dummyFunc} />, div)
  })

  test('snapshots', () => {

    const component = renderer.create(
      <Table  list={props.list} onDismiss={dummyFunc} />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})