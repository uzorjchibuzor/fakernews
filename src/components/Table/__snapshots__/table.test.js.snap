// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table snapshots 1`] = `
<div
  className="table"
>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a>
        1
      </a>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
      1
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      1
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      2
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        onClick={[Function]}
      >
        Dismiss
      </button>
    </span>
  </div>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a>
        2
      </a>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
      2
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      1
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      2
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        onClick={[Function]}
      >
        Dismiss
      </button>
    </span>
  </div>
</div>
`;
